mlrun/model_monitoring/db/tsdb/preaggregate.py

```diff
@@ -0,0 +1,221 @@
+            self.retention_policy = {
+                "raw": "7d",
+                "10m": "30d",
+                "1h": "1y",
+                "6h": "1y",
+                "1d": "5y",
+                "1w": "5y",
+                "1M": "5y",
+            }
+
+
+class PreAggregateHandler:
+    """Handles pre-aggregate validation, time alignment, and optimization decisions."""
+
+    def __init__(self, pre_aggregate_config: Optional[PreAggregateConfig] = None):
+        """
+        Initialize the pre-aggregate handler.
+
+        :param pre_aggregate_config: Configuration for pre-aggregated tables and operations.
+                                   If None, all pre-aggregate operations will be disabled.
+        """
+        self._pre_aggregate_config = pre_aggregate_config
+
+    def validate_interval_and_function(
+        self, interval: Optional[str], agg_function: Optional[str]
+    ) -> None:
+        """Validate that interval and aggregation function are available in pre-aggregate config."""
+        if not interval and not agg_function:
+            return
+
+        if not self._pre_aggregate_config:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Pre-aggregate configuration not available. Cannot use interval or agg_function parameters."
+            )
+
+        if interval and interval not in self._pre_aggregate_config.aggregate_intervals:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Interval '{interval}' not available in pre-aggregate configuration. "
+                f"Available intervals: {self._pre_aggregate_config.aggregate_intervals}"
+            )
+
+        if (
+            agg_function
+            and agg_function not in self._pre_aggregate_config.agg_functions
+        ):
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Aggregation function '{agg_function}' not available in pre-aggregate configuration. "
+                f"Available functions: {self._pre_aggregate_config.agg_functions}"
+            )
+
+    def can_use_pre_aggregates(
+        self, interval: Optional[str] = None, agg_funcs: Optional[list[str]] = None
+    ) -> bool:
+        """Check if pre-aggregates can be used for the given parameters."""
+        if not self._pre_aggregate_config or not interval:
+            return False
+
+        if interval not in self._pre_aggregate_config.aggregate_intervals:
+            return False
+
+        if agg_funcs:
+            return all(
+                func in self._pre_aggregate_config.agg_functions for func in agg_funcs
+            )
+
+        return True
+
+    def align_time_to_interval(
+        self, dt: datetime, interval: str, align_start: bool = True
+    ) -> datetime:
+        """Align datetime to interval boundaries."""
+        if not interval:
+            return dt
+
+        # Parse interval (e.g., "1h", "10m", "1d")
+        match = re.match(r"(\d+)([mhd]|M)", interval)
```

Comment:
I'm not sure we should have months as an option. A month is variable length, meaning that it's not a valid unit of duration. If we do need it (why?), here's a minor improvement to this snippet:
```suggestion
        # Parse interval (e.g., "1h", "10m", "1d", "1M")
        match = re.match(r"(\d+)([mhdM])", interval)
```